pyccel/parser/semantic.py

```diff
@@ -3814,7 +3823,7 @@ def _visit_ClassDef(self, expr):
         for method in cls.methods:
             if method.name == '__del__':
                 self._current_function = method.name
-                attribute = [attr for attr in cls.attributes if not attr.on_stack]
+                attribute = [attr for attr in cls.attributes if not attr.on_stack or isinstance(attr.dtype, CustomDataType)]
```

Comment:
Can you add parentheses to clarify this statement. Is it:
```suggestion
                attribute = [attr for attr in cls.attributes if not (attr.on_stack or isinstance(attr.dtype, CustomDataType))]
```
in which case why do we not deallocate classes?
Or is it:
```suggestion
                attribute = [attr for attr in cls.attributes if (not attr.on_stack) or isinstance(attr.dtype, CustomDataType)]
```
Is this to deallocate classes on the stack?
If so then please add a comment to clarify

Reply:
It's the first case that is correct cause if an attribute has another class as value, it shoudn't be added to the garbage collector of the current scope.

Reply:
So then when is this class attribute deallocated if not in the destructor?